src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs

```diff
@@ -48,6 +48,79 @@ public JsonArray(params JsonNode?[] items) : base()
             InitializeFromArray(items);
         }
 
+        internal JsonNode DeepCloneArray()
+        {
+            if (_jsonElement.HasValue)
+            {
+                return new JsonArray(_jsonElement.Value.Clone(), Options);
+            }
+
+            var jsonArray = new JsonArray(Options);
+
+            for (int i = 0; i < List.Count; i++)
```

Comment:
Consider storing the `List` property into a local since the getter is doing nontrivial work.